src/abstractdataframe/abstractdataframe.jl

```diff
@@ -229,23 +226,17 @@ Base.ndims(::AbstractDataFrame) = 2
 ##############################################################################
 
 Base.similar(df::AbstractDataFrame, dims::Int) =
-    DataFrame(Any[similar(x, dims) for x in columns(df)], copy(index(df)))
-
-nas{T}(dv::AbstractArray{T}, dims::Union{Int, Tuple{Vararg{Int}}}) =   # TODO move to datavector.jl?
-    DataArray(Array{T}(dims), trues(dims))
-
-nas{T,R}(dv::PooledDataArray{T,R}, dims::Union{Int, Tuple{Vararg{Int}}}) =
-    PooledDataArray(DataArrays.RefArray(zeros(R, dims)), dv.pool)
-
-nas(df::AbstractDataFrame, dims::Int) =
-    DataFrame(Any[nas(x, dims) for x in columns(df)], copy(index(df)))
+    DataFrame(Any[similar_nullable(x, dims) for x in columns(df)], copy(index(df)))
 
 ##############################################################################
 ##
 ## Equality
 ##
 ##############################################################################
 
+# Imported in DataFrames.jl for compatibility across Julia 0.4 and 0.5
+Base.:(==)(df1::AbstractDataFrame, df2::AbstractDataFrame) = isequal(df1, df2)
```

Comment:
Didn't we change this recently in DataTables? Having `==` call `isequal` defeats the purpose of having them be separate functions. If this is only for 0.4 and 0.5 compatibility, we don't support those anymore, so this could be removed.

Reply:
Yeah, I thought this looked fishy too.

Reply:
Yeah, that's also related to issues around how to define `==` for `Nullable`. We should fix this in later PRs.